esp-mbedtls/src/lib.rs

```diff
@@ -279,35 +400,58 @@ pub struct Certificates<'a> {
     /// # Server:
     /// In server mode, this will be the certificate given to the client when
     /// performing a handshake.
-    pub certificate: Option<X509<'a>>,
+    certificate: Option<MbedTLSX509Crt>,
 
     /// Private key paired with the certificate. Must be set when [Certificates::certificate]
     /// is not [None]
-    pub private_key: Option<X509<'a>>,
-
-    /// Password used for the private key.
-    /// Use [None] when the private key doesn't have a password.
-    pub password: Option<&'a str>,
+    private_key: Option<PkContext>,
 }
 
-impl Default for Certificates<'_> {
+impl Default for Certificates {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl Certificates<'_> {
+impl Certificates {
     /// Create a new instance of [Certificates] with no certificates whatsoever
     pub const fn new() -> Self {
```

Comment:
Is this a real use-case? Can the library actually operate without any certs?

Reply:
The lifetimes are covariant though so **a single lifetime would do**. If you want to keep the builder pattern, `new` would return a `Certificates<'static>` which would either stay like that (copy) or would be changed to `Certificates<'a>` by a builder non-copy method, where `'a` would be a generic lifetime capturing the actual x509 lifetime (which can _still_ end up being static after all).